### DISCUSSION TITLE
NextAuth + Prisma Local ERR_SSL_PROTOCOL_ERROR

### DISCUSSION THREAD

**Discussion by tangledbootlace** [prisma/prisma#19494] [Q&A]
Hi all!

I'm evaluating Prisma for use with a NextJS application and am following this guide https://vercel.com/guides/nextjs-prisma-postgres. However, I am running into issues on Step 7.

After setting up my /api/auth/ route, I successfully see the page with the button to "Sign in with GitHub," however, upon clicking the button, I am redirected to https://localhost:3000/api/auth/signin/github and get an ERR_SSL_PROTOCOL_ERROR due to the fact that I do not have SSL configured.

I've followed this guide incredibly closely, and am not sure if I am missing something. Has anyone else run into this issue while following this guide?

Any help would be appreciated! I'm happy to provide any additional information needed.

**Comment by nurul3101** [comment]
Hello @tangledbootlace 👋 

Thanks for raising this, it's on our team's radar to update the Guide.
Here's a related GitHub Discussion discussing errors for step 7 and 8
- https://github.com/prisma/prisma/discussions/19276

Can you have a look at this GitHub Discussion?

**Reply by tangledbootlace** [reply]
Hey there!

Took a look at that discussion, and it seems to be focused on issues around step 8. I'm specifically getting an error at the very end of Step 7. Do you know of any solution for the ERR_SSL_PROTOCOL_ERROR related to the redirect after clicking the "Sign In With Github" button?

**Reply by nurul3101** [reply]
So if I understand correctly, instead of getting redirected to GitHub you are getting the `ERR_SSL_PROTOCOL_ERROR` error, right?

**Reply by tangledbootlace** [reply]
Correct, instead of landing on the github login page, I get the ERR_SSL_PROTOCOL_ERROR. At first I thought this was due to already being logged into Github and the redirect was sending me to http, but I've confirmed my Authorization callback URL is http://localhost:3000/api/auth and the same ERR_SSL_PROTOCOL_ERROR occurs whether I'm logged in or not

**Reply by nj606214** [reply]
@nurul3101 @tangledbootlace please save us!! I am having this same issue when trying to sign in with google and have been banging my head against the wall for days.

**Reply by nurul3101** [reply]
@nj606214 which version of Next Auth are you using? Can you please try with version `4.22.0`?

**Comment by ezegonmac** [comment]
I am having the same issue but for credentials provider.

When I click the "Sign in with Credentials" button from the default NextAuth page I am redirected to "https://localhost:3000/api/auth/callback/credentials" but it shows this error "ERR_SSL_PROTOCOL_ERROR This Site Can’t Provide a Secure Connection".

The same happens when I click "Sign Out" in the signout page.

I tried disabling certificate verification with the env variable "NODE_TLS_REJECT_UNAUTHORIZED=0" I have seen in this answer https://stackoverflow.com/a/58003959/11909334

**Reply by ezegonmac** [reply]
Finally I found the error after a lot of hours of searching... I had my NEXTAUTH_URL defined as "https://localhost:3000" with HTTPS instead of HTTP. 

@tangledbootlace maybe your problem could be something similar.

**Reply by nurul3101** [reply]
Thanks for getting back and letting us know! 😄

**Reply by nurul3101** [reply]
@tangledbootlace Let us know in case this fixes the issue for you?

**Reply by tangledbootlace** [reply]
I double checked, and I have the correct http:// address in my .env file

**Comment by gertoska** [comment]
I was experiencing exactly the same issue. Even having the variable env `NEXTAUTH_URL="http://..."` was still redirecting to https://. Check if in your .env file you have `VERCEL="1"`, if so it always redirects to https. Put `VERCEL=""` and it should work.

Caused by https://github.com/nextauthjs/next-auth/blob/main/packages/next-auth/src/utils/detect-origin.ts
```
if (process.env.VERCEL ?? process.env.AUTH_TRUST_HOST)
    return `${protocol === "http" ? "http" : "https"}://${forwardedHost}`
```

This post from StackOverflow helped me: https://stackoverflow.com/a/76104275

**Reply by mathiasrscom** [reply]
I can confirm that changing to ```VERCEL=""``` fixed it for me.

**Reply by imkhoirularifin** [reply]
Thank you, this is fixed my problem

**Comment by badace7** [comment]
VERCEL="" fixed my problem. Thx.

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
